plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -240,11 +250,25 @@ func compareTopics(expected []string, incoming []string) bool {
 }
 
 func (m *MQTTConsumer) onMessage(acc telegraf.TrackingAccumulator, msg mqtt.Message) error {
+	m.bytesRecv.Set(0)
 	metrics, err := m.parser.Parse(msg.Payload())
 	if err != nil {
 		return err
 	}
 
+	if m.collectBytes {
```

Comment:
I would place this before parsing the payload, so you can have these metrics even if the parsing fails..